sudoku_board.cc

```diff
@@ -0,0 +1,123 @@
+
+#include "sudoku_board.h"
+
+// ------------------------------- public methods ---------------------------------
+
+SudokuBoard::SudokuBoard(int size){
+    m_size = size;
+    m_board_values = std::vector(size, std::vector<int>(size, EMPTY_CELL_SIGN));
+    m_possible_options = std::vector(size, std::vector(size, std::set<int>(POSSIBLE_CELL_VALUES)));
+}
+
+SudokuBoard::SudokuBoard(int size, std::vector<std::vector<int>> board_values){
+    m_size = size;
+    m_board_values = board_values; //question: when to use a copy constructor?
+    m_possible_options = std::vector(size, std::vector(size, std::set<int>(POSSIBLE_CELL_VALUES)));
+    update_possible_options();
+}
+
+int SudokuBoard::get_size(){
+    return m_size;
+}
+
+// question: do I need to return a copy and not the original one?
```

Comment:
Answer: what is your use case? Do you really want to copy the entire board every single time (this is the current case)? Perhaps sometimes you just want to reference it (like, just kinda read from it...) and then it's wasteful to copy